Tests/WorkspaceTests/InitTests.swift

```diff
@@ -14,458 +14,422 @@ import Basics
+            })
         }
     }
 
-    func testInitPackageExecutableWithSwiftTesting() async throws {
-        try testWithTemporaryDirectory { tmpPath in
-            let fs = localFileSystem
-            let path = tmpPath.appending("Foo")
-            let name = path.basename
-            try fs.createDirectory(path)
-            // Create the package
-            let initPackage = try InitPackage(
-                name: name,
-                packageType: .executable,
-                supportedTestingLibraries: [.swiftTesting],
-                destinationPath: path,
-                fileSystem: localFileSystem
-            )
-
-            try initPackage.writePackageStructure()
-            // Verify basic file system content that we expect in the package
-            let manifest = path.appending("Package.swift")
-            XCTAssertFileExists(manifest)
-            let manifestContents: String = try localFileSystem.readFileContents(manifest)
-            XCTAssertMatch(manifestContents, .contains(".testTarget("))
-            let testFile = path.appending("Tests").appending("FooTests").appending("FooTests.swift")
-            let testFileContents: String = try localFileSystem.readFileContents(testFile)
-            XCTAssertMatch(testFileContents, .contains(#"import Testing"#))
-            XCTAssertNoMatch(testFileContents, .contains(#"import XCTest"#))
-            XCTAssertMatch(testFileContents, .contains(#"@Test func example() async throws"#))
-            XCTAssertNoMatch(testFileContents, .contains("func testExample() throws"))
-        }
+    /// Tests creating an executable package with different build systems.
+    @Test(arguments: [BuildSystemProvider.Kind.native, .swiftbuild])
+    func initPackageExecutable(buildSystem: BuildSystemProvider.Kind) async throws {
+        try await createAndVerifyPackage(
+            packageType: .executable,
+            buildSystem: buildSystem,
+            customVerification: { path, name in
+                let directoryContents = try localFileSystem.getDirectoryContents(path.appending("Sources").appending(name))
+                #expect(directoryContents == ["\(name).swift"])
+            }
+        )
     }
 
-    func testInitPackageToolWithSwiftTesting() async throws {
-        try testWithTemporaryDirectory { tmpPath in
-            let fs = localFileSystem
-            let path = tmpPath.appending("Foo")
-            let name = path.basename
-            try fs.createDirectory(path)
-            // Create the package
-            let initPackage = try InitPackage(
-                name: name,
-                packageType: .tool,
-                supportedTestingLibraries: [.swiftTesting],
-                destinationPath: path,
-                fileSystem: localFileSystem
-            )
+    /// Tests creating an executable package named "main".
+    @Test(arguments: [BuildSystemProvider.Kind.native, .swiftbuild])
```

Comment:
suggestion: consider replacing the value of `arguments` with  `SupportedBuildSystemOnAllPlatforms`.  This comment applies to all occurrences of an array of two items with `native` and `swiftbuild` build systems.